app/lib/actions/katello/repository/destroy_package_group.rb

```diff
@@ -7,7 +7,8 @@ def plan(repository, pkg_group_id)
           criteria = { type_ids: ["package_group"], filters: {"unit": {"name": pkg_group_id} } }
 
           sequence do
-            ::Katello.pulp_server.extensions.repository.unassociate_units(repository.pulp_id, criteria)
+            # TODO: Replace with Pulp 3?
+            fail 'Pulp 2 was used'
             plan_action(IndexPackageGroups, repository)
             plan_action(Katello::Repository::MetadataGenerate, repository)
```

Comment:
Rubocop failures are here, but really the question is whether this action is still used. It's called here from a REST API controller which suggests it can still be called by users:
https://github.com/Katello/katello/blob/b056c7960fe99f709cb225c68cd1a321ec105840/app/controllers/katello/api/v2/package_groups_controller.rb#L54

Reply:
We don't support deleting package groups from the UI or Hammer, so I think it's safe to just drop it.

Re-implementing it would be relatively easy, but it would need to be tested and such. We'd use the remove_content API endpoint for repositories. The related tasks can accept any sort of content unit, but package groups would need to be added as an acceptable type.